src/player_activity.cpp

```diff
@@ -61,38 +131,36 @@ std::string player_activity::get_str_value( size_t index, const std::string &def
 
 cata::optional<std::string> player_activity::get_progress_message() const
 {
+    const auto iter = activity_names.find( type );
+    if( iter == activity_names.end() ) {
+        return cata::optional<std::string>();
+    }
+
+    std::string extra_info = "";
     if( type == activity_id( "ACT_CRAFT" ) ) {
         if( const item *craft = targets.front().get_item() ) {
-            return string_format( _( "Crafting: %s" ), craft->tname() );
+            extra_info = craft->tname();
         }
     } else if( moves_total > 0 ) {
         const int percentage = ( ( moves_total - moves_left ) * 100 ) / moves_total;
 
-        if( type == activity_id( "ACT_BURROW" ) ) {
-            return string_format( _( "Burrowing: %d%%" ), percentage );
-        } else if( type == activity_id( "ACT_HACKSAW" ) ) {
-            return string_format( _( "Sawing: %d%%" ), percentage );
-        } else if( type == activity_id( "ACT_JACKHAMMER" ) ) {
-            return string_format( _( "Jackhammering: %d%%" ), percentage );
-        } else if( type == activity_id( "ACT_PICKAXE" ) ) {
-            return string_format( _( "Digging: %d%%" ), percentage );
-        } else if( type == activity_id( "ACT_DISASSEMBLE" ) ) {
-            return string_format( _( "Disassembling: %d%%" ), percentage );
-        } else if(
+        if( type == activity_id( "ACT_BURROW" ) ||
+            type == activity_id( "ACT_HACKSAW" ) ||
+            type == activity_id( "ACT_JACKHAMMER" ) ||
+            type == activity_id( "ACT_PICKAXE" ) ||
+            type == activity_id( "ACT_DISASSEMBLE" ) ||
             type == activity_id( "ACT_FILL_PIT" ) ||
             type == activity_id( "ACT_DIG" ) ||
-            type == activity_id( "ACT_DIG_CHANNEL" )
-        ) {
-            return string_format( _( "Shoveling: %d%%" ), percentage );
-        } else if(
+            type == activity_id( "ACT_DIG_CHANNEL" ) ||
             type == activity_id( "ACT_CHOP_TREE" ) ||
             type == activity_id( "ACT_CHOP_LOGS" ) ||
             type == activity_id( "ACT_CHOP_PLANKS" )
-        ) {
-            return string_format( _( "Chopping: %d%%" ), percentage );
+          ) {
+            extra_info = string_format( "%d%%", percentage );
         }
     }
-    return cata::optional<std::string>();
+
+    return string_format( "%s%s", _( iter->second ), extra_info.empty() ? "…" : ": " + extra_info );
```

Comment:
This might be better for translators:
```c++
    return extra_info.empty() ? string_format( "%s…", _( iter->second ) ) : string_format( "%s: %s", _( iter->second ), extra_info );
```

Reply:
I mean `_( "%s…" )` and `_( "%s: %s" )`.

Reply:
Does it really need translating through?

Reply:
I'm not sure. Probably not.

Reply:
In Chinese translation we need to substitute colon+space with the full-width colon, so it'll be better translated. Also you don't need to append  the percentage this way. Instead you can just place the ellipsis and `: %d%%` into the respective activity names, and call `string_format` using the name and the percentage like `string_format( _( iter->second ), percentage );`. If the name does not contain `%d%%`, the percentage will just be ignored.